pkg/tools/serialize/serialize.go

```diff
@@ -19,18 +19,22 @@
 // will be executed in order
 package serialize
 
-import "runtime"
+import (
+	"sync"
+)
 
 // Executor - allows one at a time in order execution of func()s.  func()s can be queued Asynchronously or
 // Synchronously
 type Executor interface {
 	AsyncExec(func())
 	SyncExec(func())
+	Stop()
```

Comment:
Why Stop() rather than just the finalizer?

Reply:
Fixed with `per-chain` Context